discord/poll.py

```diff
@@ -0,0 +1,624 @@
+
+        if not self._message:
+            return
+        return self._message.created_at
+
+    @property
+    def message(self) -> Optional[Message]:
+        """:class:`Message`: The message this poll is from."""
+        return self._message
+
+    @property
+    def total_votes(self) -> int:
+        """:class:`int`: Returns the sum of all the answer votes."""
+        return sum([answer.vote_count for answer in self.answers])
+
+    def is_finalized(self) -> bool:
+        """:class:`bool`: Returns whether the poll has finalized.
+
+        It always returns ``False`` for stateless polls.
+        """
+        return self._finalized
+
+    is_finalised = is_finalized
+
+    def copy(self) -> Self:
+        """Returns a stateless copy of this poll.
+
+        This is meant to be used when you want to edit a stateful poll.
+
+        Returns
+        -------
+        :class:`Poll`
+            The copy of the poll.
+        """
+
+        new = self.__class__(question=self.question, duration=self.duration)
+        # We want to return a stateless copy of the poll, so we should not
+        # override new._answers as our answers may contain a state
+
+        for answer in self.answers:
+            new.add_answer(text=answer.text, emoji=answer.emoji)
+
+        return new
+
+    def add_answer(
+        self,
+        *,
+        text: str,
+        emoji: Optional[Union[PartialEmoji, Emoji, str]] = None,
+    ) -> Self:
+        """Appends a new answer to this poll.
+
+        Parameters
+        ----------
+        text: :class:`str`
+            The text label for this poll answer. Can be up to 55
+            characters.
+        emoji: Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]
+            The emoji to display along the text.
+
+        Raises
+        ------
+        RuntimeError
+            Cannot append answers to a poll recieved via message.
+
+        Returns
+        -------
+        :class:`Poll`
+            This poll with the new answer appended.
+        """
+
+        if self._message:
+            raise RuntimeError('Cannot append answers to a poll recieved via message.')
+
+        answer = PollAnswer.from_params(id=len(self.answers) + 1, text=text, emoji=emoji, message=self._message, poll=self)
+
+        self._answers.append(answer)
+        return self
+
+    def get_answer(
+        self,
+        /,
+        id: int,
+    ) -> Optional[PollAnswer]:
+        """Returns the answer with the provided ID or ``None`` if not found.
+
+        Note that the ID, as Discord says, it is the index / row where the answer
+        is located in the poll.
+
+        Parameters
+        ----------
+        id: :class:`int`
+            The ID of the answer to get.
+
+        Returns
+        -------
+        Optional[:class:`PollAnswer`]
+            The answer.
+        """
+
+        return utils.get(self.answers, id=id)
```

Comment:
You ignored this earlier but `get_` methods **should not** be O(n). Either change it to a dictionary or remove the method entirely.